src/promnesia/sources/viber_desktop.py

```diff
@@ -0,0 +1,164 @@
+                CR.PGRole,
+                CI.PGTags
+            FROM ChatRelation as CR
+            JOIN Contact AS C ON CR.ContactID = C.ContactID
+            JOIN ChatInfo as CI ON CR.ChatId = CI.ChatId
+        ), G1 AS (
+            SELECT * FROM G0 WHERE PGRole = 2
+        ), G2 AS (
+            SELECT * FROM G0 WHERE
+                ContactID <> 1 AND
+                ChatId NOT IN (SELECT ChatId FROM G1)
+        ), Groups AS (
+            SELECT ChatId, chatname, PGTags FROM G1
+            UNION
+            SELECT ChatId, chatname, PGTags FROM G2
+        )
+        SELECT
+            M.EventId           AS mid,
+            E.TimeStamp         AS time,
+            G.chatname          AS chatname,
+            coalesce(
+                S.Name,
+                S.ClientName,
+                '(' || S.Number || ')'          /* contacts have one xor the other, but failsafe */
+            )                   AS sender,
+            coalesce(M.Subject, M.Body)         /* didn't see any msg with both */    
+                                AS text,
+            M.info              AS infojson,    /* to harvested titles from embedded urls */
+            G.PGTags            AS tags
+        FROM messages AS M
+        LEFT JOIN Events AS E
+            ON M.EventId = E.EventId
+        LEFT JOIN Contact AS S
+            ON E.ContactId = S.ContactId
+        LEFT JOIN Groups AS G
+            ON E.ChatId = G.ChatId
+        WHERE
+            text LIKE '%http%'
+        ORDER BY time;
+        """
+    )
+
+    def _parse_json_title(js) -> str:
+        if js and js.strip():
+            js = json.loads(js)
+            if isinstance(js, dict):
+                return js.get("Title")
+
+    def _handle_row(row) -> Results:
+        text = row["text"]
+        assert (
+            text
+        ), f"sql-query should have eliminated messages not containing 'http': {text}"
+        urls = extract_urls(text)
+        if not urls:
+            return
+        dt = from_epoch(row["time"] // 1000)  # timestamps are stored x100 this db
+        mid: str = unwrap(row["mid"])
+        # TODO perhaps we could be defensive with null sender/chat etc and still emit the Visit
+        sender: str = unwrap(row["sender"])
+        chatname: str = unwrap(row["chatname"])
+        sender: str = unwrap(row["sender"])
+        tags: str = unwrap(row["tags"])
+        infojson: str = unwrap(row["infojson"])
+
+        if tags and tags.strip():
+            tags = "".join(f"#{t}" for t in tags.split())
+            text = f"{text}\n\n{tags}"
+
+        url_title = _parse_json_title(infojson)
+        if url_title:
+            text = f"title: {url_title}\n\n{text}"
+
+        for u in urls:
+            # https://www.reddit.com/r/Telegram/comments/6ufwi3/link_to_a_specific_message_in_a_channel_possible/
+            # hmm, only seems to work on mobile app, but better than nothing...
+            yield Visit(
+                url=unquote(u),
```

Comment:
are they quoted in the viber database? Might be unnecessary (and if they are do you mind adding a comment?)

Reply:
No they are not, removed it.
Resolved in f4553ed.